File: 121-Best-Time-to-Buy-and-Sell-Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l,r=0,1
        max_profit=0
        while r<len(prices):
            profit=prices[r]-prices[l]
            max_profit=max(profit,max_profit)
            if profit <0:
                l=r
                r=l+1
            else:
                r+=1
        return max_profit

File: Top_interview_150/121-Best-Time-to-Buy-and-Sell-Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int buy=prices[0],sell=prices[0];
        int max_profit=0;

        for(int price : prices){
            if(price<buy){
                buy=price;
                sell=price;
            }
            else{
                sell=price;
            }
            int profit=sell-buy;
            max_profit=max(max_profit,profit);
        }
        return max_profit;

    }
};

